prov/sockets/src/sock_ep_msg.c

```diff
@@ -822,6 +822,9 @@ static int sock_ep_cm_connect(struct fid_ep *ep, const void *addr,
 	if (!_ep->attr->listener.listener_thread && sock_conn_listen(_ep->attr))
 		return -FI_EINVAL;
 
+	if (!addr)
+		return -FI_EINVAL;
+
```

Comment:
Please merge this with the if statement above the one above.  :)  I.e. the one that checks for !_eq.